Capstone Project | Twitter clone

miankhalid opened the pull request:
# Project Summary

## Overall Goal
Implement a production-grade Threads/Twitter/X clone using React Native CLI (v0.80.2) with Redux Toolkit, React Query, Axios, Shopify Restyle, MMKV, Supabase, Zod, and other specified technologies following the existing project architecture.

## Key Knowledge
### Technology Stack
- **Frontend**: React Native (CLI only, no Expo), TypeScript optionally, JavaScript when appropriate
- **State Management**: Redux Toolkit for UI state, React Query (v5) for server state
- **API**: Axios as HTTP client
- **Theming**: Shopify Restyle for theming and dark mode
- **Storage**: MMKV for fast key-value storage
- **Backend**: Supabase for Auth/Postgres/Realtime with RLS and secure token handling
- **Validation**: Zod for runtime validation
- **Other**: Reanimated, i18next, ESLint, Prettier, commitlint, husky, lint-staged, Jest

### Project Structure Constraints
- Follow existing `src/` architecture with components/atoms, molecules, organisms, templates
- hooks/, navigation/, screens/, services/, theme/, translations/, utils/
- Use Conventional Commits for all commits

### Development Constraints
- No Expo, React Native CLI only
- Target RN 0.80.2
- Mix of TypeScript and JavaScript allowed where appropriate
- Follow existing project folder structure exactly
- Use `/Kryos/sample/ThreadsClone` for design inspiration only, not code copying
- Use atomic commits to current feature branch only

### Architectural Decisions
- Shopify Restyle for theming system with dark mode toggle
- MMKV for token persistence
- Supabase client wrapper with token refresh logic
- React Query with Axios integration
- Redux Toolkit for authentication state management
- Zod validation schemas for all API responses

miankhalid committed:
feat(theme): configure shopify restyle for theming and dark mode

miankhalid committed:
feat(store): integrate Redux Toolkit for UI state management

- Installs `@reduxjs/toolkit` and `react-redux`.
- Configures the main Redux store, including middleware to disable serializableCheck for compatibility with non-serializable data.
- Creates a `uiSlice` to manage global UI state, starting with a `isSystemLoading` flag.
- Defines and exports typed `useAppDispatch` and `useAppSelector` hooks for type-safe interaction with the store.
- Wraps the root `App` component with the Redux `Provider` to make the store available throughout the application.
- Adds an `index.ts` barrel file to the `store` directory to resolve a Metro bundler module resolution error.

miankhalid committed:
feat(services): Configure axios for http requests

Installs axios and creates a reusable, pre-configured instance to be
used for all API calls.

The new instance is located in src/services/instance.ts and includes
a placeholder for the API baseURL. It also contains commented-out
interceptor code to streamline adding auth tokens to requests later.

miankhalid committed:
feat(services): Integrate Supabase client and configure Axios

Installs the Supabase client and configures it for authentication.
Updates the existing Axios instance to use the Supabase REST API
endpoint and credentials from environment variables.

A new supabase.ts service initializes the client, and a type
declaration file (env.d.ts) is added to ensure type-safety for the
environment variables used by both Supabase and Axios, resolving
linting errors.

miankhalid committed:
feat(storage): Integrate MMKV as Supabase auth persistor

Creates a shared, centralized MMKV instance and a custom storage
adapter. The Supabase client is then configured to use this adapter,
enabling secure, performant, and persistent session storage for
authentication.

This replaces the default browser-based localStorage, which is not
available in React Native, and refactors the app to use a single,
shared storage instance for better code organization.

miankhalid committed:
feat(auth): Build authentication UI screens and components

Creates the user interface for the Login and Signup screens. This
includes new reusable, themed Input and Button components that are
built using the Shopify Restyle system.

The new components are used to construct the forms for user login and
registration. The restyleTheme is updated to include variants for the
new Button component, ensuring a consistent design.

miankhalid committed:
refactor(theme): Remove old theming boilerplate

Deleted unused and not-needed theme resources, including old configuration files,
types, and components. This cleanup addresses lingering issues related to the
previous theming system and ensures a cleaner, Restyle-only theme implementation.

miankhalid committed:
feat(auth): Implement authentication logic and session handling

Implements the core authentication logic by creating a useAuth hook
that subscribes to Supabase's auth state changes. This hook provides
the user's session and loading status throughout the app.

The root navigator is updated to use this hook, automatically showing
the correct screen (Startup, Auth, or Main) based on the session
status. The Login, Signup, and Home screens are wired up to the new
login, signup, and logout functions.

miankhalid committed:
refactor!: remove unused Startup screen and related configs

- Remove unused Startup screen component and its navigation entries
- Clean up theme asset and configuration files that are no longer
needed
- Update ESLint configuration to use flat config format with improved
organization
- Fix Reactotron config to use correct storage import
- Update import sorting rules to include services and utils paths
- Add sample directory to ESLint ignore list

These changes simplify the codebase by removing unused components and
updating configurations to modern standards

miankhalid committed:
refactor!: remove unused Startup screen and related configs

- Remove unused Startup screen component and its navigation entries
- Clean up theme asset and configuration files that are no longer
needed
- Update ESLint configuration to use flat config format with improved
organization
- Fix Reactotron config to use correct storage import
- Update import sorting rules to include services and utils paths
- Add sample directory to ESLint ignore list

These changes simplify the codebase by removing unused components and
updating configurations to modern standards

miankhalid committed:
feat: implement client-side validation with i18n support for auth forms

- Added Zod schemas for login and signup form validation
- Implemented form validation using react-hook-form with zodResolver
- Integrated i18n for validation error messages with support for English and French
- Updated Login and Signup screens with proper error display and validation
- Added new translation keys for auth-related error messages
- Ensured validation errors are shown in real-time as users type

The changes provide comprehensive client-side validation for authentication
forms with localized error messages.

miankhalid committed:
feat(auth): Add loading states to auth screens

Implement loading indicators on the Login and Signup screens to provide
better user feedback during authentication requests.

This change introduces:
- A new `authSlice` to manage authentication-related state, including
`isLoading` and `error`.
- Integration of `authSlice` into the Redux store configuration.
- Modifications to `LoginScreen` and `SignupScreen` to dispatch `setLoading`
actions before and after API calls.
- Utilization of the `loading` prop on the `Button` component to display an
`ActivityIndicator`.
- A fix for an issue where the `authReducer` was incorrectly referenced in
`store.ts`, preventing the `isLoading` state from being accessed.

miankhalid committed:
feat(feed): implement feed screen with auth navigation

- Create Feed screen with dummy data display using FlatList
- Add auth panel with Login and Signup buttons at bottom of
feed
- Implement proper navigation flow between Feed, Login, and
Signup screens
- Add SafeAreaView to handle notches and Dynamic Island
properly
- Update navigation flow: non-logged users see Feed first,
logged users see Home
- Add logout functionality that returns users to Feed screen
- Ensure all screens have proper navigation contexts

miankhalid committed:
feat: Add profile pictures to feed posts

- Create Image component with Shopify Restyle support
- Update FeedScreen to display user profile images from
dummy data
- Use default avatar placeholder as fallback
- Improve feed UI with profile pictures next to each post

miankhalid commented:
Video of progress until now 👇 

https://github.com/user-attachments/assets/289e11fe-2855-47a7-aacb-af87a0a95f19

usmankhalid81 commented on react-native/Kryos/src/components/atoms/Button/Button.tsx:
```diff
@@ -0,0 +1,104 @@
+import Text from '@/components/atoms/Text';
+
+// Constants
+const ACTIVE_OPACITY = 0.7;
+const DISABLED_OPACITY = 0.5;
+const SPINNER_SIZE = 20;
+
+// Button Props
+export type ButtonProps = {
+  containerProps?: React.ComponentProps<typeof Box>;
+  disabled?: boolean;
+  loading?: boolean;
+  rightIcon?: React.ReactNode;
+  title: string;
+  variant?: 'primary' | 'text';
+} & TouchableOpacityProps;
```
For now,  it’s good to have. It’s preferable to maintain constants and types in separate relevant files.

usmankhalid81 commented on react-native/Kryos/src/components/atoms/Input/Input.tsx:
```diff
@@ -0,0 +1,180 @@
+} & Omit<RNTextInputProps, 'placeholderTextColor'>;
+
+// Base TextInput Component
+export const Input = forwardRef<RNTextInput, InputProps>(
```
please explore  the usage of forwardRef.

usmankhalid81 commented on react-native/Kryos/src/hooks/auth/useAuth.ts:
```diff
@@ -0,0 +1,62 @@
+import { supabase } from '@/services/supabase';
+
+export const login = async (credentials: SignInWithPasswordCredentials) => {
+  const { data, error } = await supabase.auth.signInWithPassword(credentials);
```
We are using the SDK instead of hitting the API endpoint?

usmankhalid81 commented on react-native/Kryos/src/schemas/authSchema.ts:
```diff
@@ -0,0 +1,38 @@
+import { authErrorKeys } from '@/constants/authErrorMessages';
+
+// Email validation regex for more specific email format checking
+const emailRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
```
Instead of creating a custom regex, Zod must have an internal email validator that can be called using, i.e., .email().

usmankhalid81 commented on react-native/Kryos/src/screens/Auth/LoginScreen.tsx:
```diff
@@ -0,0 +1,134 @@
+
+  const handleLogin = async (data: LoginFormData) => {
+    dispatch(setLoading(true));
+    dispatch(clearError());
```
why are we clearing error? in case of validation error handleLogin shouldn’t be called.

usmankhalid81 commented on react-native/Kryos/src/screens/Auth/SignupScreen.tsx:
```diff
@@ -0,0 +1,155 @@
+          Alert.alert(
+            'Signup Success',
+            'Please check your email to confirm your account.',
+          );
```
for batter UX. can we show snackbar?

usmankhalid81 reviewed (commented):
Great. It’s a large PR. I just reviewed some files and will continue reviewing this.

miankhalid committed:
refactor(types): centralize types and refactor navigation paths

- Move types to dedicated types/types.ts file
- Update dummy data to import from new types location
- Refactor navigation paths to use enum in paths.ts
- Update AuthNavigator, LoginScreen, and other navigation references to use Paths enum
- Update navigation types to include new paths
- Update domain index to export post hooks module
- Improve navigation type safety with enum-based paths

miankhalid committed:
feat(post): implement new post creation functionality

- Create NewPostScreen component with form and submission
- Implement posts service with REST API functions for create/fetch operations
- Add authentication token handling to API instance interceptor
- Create authAPI service for token management
- Add post validation schemas using Zod
- Create React Query hooks for posts (usePosts, useCreatePost)
- Update HomeScreen to include new post button
- Add API endpoints centralization file
- Connect NewPostScreen with authentication state to get current user
- Update MainNavigator to include NewPost route

miankhalid committed:
feat(navigation): add bottom tab navigation

- Add TabNavigator with BottomTabs from @react-navigation/bottom-tabs
- Create Settings screen with logout functionality
- Add Create screen as placeholder component
- Update navigation structure to use tab-based navigation for
authenticated users
- Add internationalization keys for tab titles in en-EN.json
- Configure vector icons library for tab icons
- Update navigation types to support tab-based structure
- Fix Android build issues related to new architecture configuration

miankhalid committed:
chore(lint): apply automated linting fixes across codebase

- Email validation using Zod's internal function
- Fix import ordering to meet linter requirements
- Correct type names (e.g., MainStakParamList → MainStackParamList)
- Standardize component and variable naming conventions
- Apply code formatting fixes to meet style guide
- Resolve dependency and path import issues
- Fix minor type inconsistencies
- Address unused import warnings
- Apply consistent formatting across all components and services

miankhalid committed:
fix(i18n): ensure i18n is initialized at an app level

miankhalid committed:
feat: enhance Settings screen with profile editing

- Add profile editing form with username, full_name, avatar_url and
website fields
- Implement auto-population of fields with existing Supabase user data
- Add email field that is visible but not editable
- Connect form to Supabase update functionality
- Add placeholder with user initials when no avatar is available
- Create utilities directory with textUtilities.ts containing reusable
getInitials function

miankhalid commented:
Newly implemented stuff:
1. Bottom tab bars after login
2. Create a post tab
3. Settings tab with profile update features

Video of progress until now 👇

https://github.com/user-attachments/assets/c58a8057-2117-4fee-8346-0106e2b4795a

miankhalid committed:
feat: implement Feeds screen using Supabase's REST API

- Create postService using axios instance with Supabase REST API
- Implement React Query hooks for posts data fetching and management
- Update FeedsScreen to use useQuery for automatic data fetching
- Add proper loading and error handling states
- Reuse existing UI functionality with avatar placeholders and post
rendering
- Implement background refetching every 5 minutes

miankhalid committed:
fix(eslint): resolve android build & linting issues

- Fixed 'react/require-default-props' errors by adding default values
to optional component props
- Resolved 'no-floating-promises' issues in React Query hooks by
properly handling async operations
- Fixed 'no-unsafe' errors by adding proper TypeScript typing and
response typing
- Replaced logical OR operators with nullish coalescing operators for
safer default assignments
- Fixed enum comparison errors in navigation components
- Removed unused imports and variables
- Improved error handling with proper error typing in services
- Updated dependency versions for react-native packages
- Removed problematic eslint override causing conflicts

miankhalid committed:
feat(posts): Implement posts feed, detail view, and creation

- Posts Feed: Implemented PostItem component and usePostsQuery hook to
display a feed of posts.
- Post Detail View: Added PostDetailScreen to show individual post
details.
- Post Creation: Updated NewPostScreen and related services to support
creating new posts.
- UI Enhancements: Integrated Avatar component and made various UI
adjustments across screens.
- Data Management: Refactored post-related hooks and services for
better data fetching and state management.

usmankhalid81 commented on react-native/Kryos/src/App.tsx:
```diff
@@ -2,12 +2,16 @@ import 'react-native-gesture-handler';
+
 export const queryClient = new QueryClient({
   defaultOptions: {
     mutations: {
```
Let'e move this in separate file with default configuration .

usmankhalid81 commented on react-native/Kryos/src/App.tsx:
```diff
@@ -19,16 +23,16 @@ export const queryClient = new QueryClient({
-        </ThemeProvider>
-      </QueryClientProvider>
+    <GestureHandlerRootView style={{ flex: 1 }}>
+      <Provider store={store}>
```
Let’s rename it to ReduxProvider because this is related to Redux. The reason to use a relevant name is that while reviewing, I don’t know what this provider indicates I have to check the import.

usmankhalid81 commented on react-native/Kryos/src/hooks/auth/useAuth.ts:
```diff
@@ -0,0 +1,70 @@
+  return data;
+};
+
+export const signup = async (credentials: SignUpWithPasswordCredentials) => {
+  const { data, error } = await supabase.auth.signUp(credentials);
+  if (error) {
+    throw new Error(error.message);
+  }
+  return data;
+};
+
+export const logout = async () => {
```
These services  can be moved to auth service.

usmankhalid81 commented on react-native/Kryos/src/hooks/auth/useAuth.ts:
```diff
@@ -0,0 +1,70 @@
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const getSession = async () => {
+      if (!supabase) {
+        console.error('Supabase client is not initialized.');
+        setLoading(false);
+        return;
+      }
+      const { data } = await supabase.auth.getSession();
+      setSession(data.session ?? undefined);
+      setLoading(false);
+    };
+
+    void getSession();
+
+    if (!supabase) {
+      console.error(
+        'Supabase client is not initialized, cannot set up auth listener.',
+      );
+      return;
+    }
+    const { data: authListener } = supabase.auth.onAuthStateChange(
+      (_event, session) => {
+        setSession(session ?? undefined);
+      },
+    );
```
Considering SOLID Principle.  Let's move this in auth-session-service to manage session related thing i.e getSession, setSessionChange etc

usmankhalid81 commented on react-native/Kryos/src/hooks/auth/useAuthWithProfile.ts:
```diff
@@ -0,0 +1,40 @@
+        }
+      } else {
+        // Clear profile when no session
+        dispatch(clearProfile());
```
Wouldn’t clearProfile be  executed even user session is loading?

usmankhalid81 commented on react-native/Kryos/src/hooks/auth/useAuthWithProfile.ts:
```diff
@@ -0,0 +1,40 @@
+  const { loading: profileLoading, profile } = useAppSelector((state) => state.profile);
+
+  useEffect(() => {
+    const loadProfile = async () => {
```
>ove this function outside of useEffect and use useCallback if required.

usmankhalid81 commented on react-native/Kryos/src/hooks/domain/post/postHooks.ts:
```diff
@@ -0,0 +1,44 @@
+export const usePosts = () => {
+  return useQuery<Post[], Error>({
+    queryFn: async () => await fetchPostsFromRestAPI(),
+    queryKey: ['posts'],
```
queryKey: [QUERY_KEYS.POSTS]
please use Named constant.

usmankhalid81 commented on react-native/Kryos/src/hooks/domain/post/postHooks.ts:
```diff
@@ -0,0 +1,44 @@
+    mutationFn: (postData: CreatePostData) => createPostWithRestAPI(postData),
+    onSuccess: () => {
+      // Invalidate and refetch posts
+      void queryClient.invalidateQueries({ queryKey: ['posts'] });
```
same as above and update for all occurrence

usmankhalid81 commented on react-native/Kryos/src/navigation/TabNavigator.tsx:
```diff
@@ -0,0 +1,64 @@
+        headerShown: false,
+        tabBarActiveTintColor: colors.bgPrimary,  // primary color from theme
+        tabBarIcon: ({ color, focused, size }) => {
+          const iconMap: Record<keyof TabParamList, string> = {
+            Create: focused ? 'add-card' : 'add-card',
+            Feeds: focused ? 'forum' : 'forum',
+            Settings: focused ? 'account-circle' : 'account-circle',
+          };
+          const iconName = iconMap[route.name] || 'help';
```
Such logic should be moved to utils.

usmankhalid81 commented on react-native/Kryos/src/screens/Auth/SignupScreen.tsx:
```diff
@@ -0,0 +1,162 @@
+  };
+
+  return (
+    <Box flex={1} justifyContent="center" padding="xl">
```
Component should be divided as much as possible.This component can be divided this component to header, body, footer components.

usmankhalid81 commented on react-native/Kryos/src/screens/Feed/DummyFeedScreen.tsx:
```diff
@@ -0,0 +1,100 @@
+import avatarPlaceholder from '../../theme/assets/images/default_avatar.png';
+
+function FeedScreen({ navigation }: { navigation: AuthScreenProps<'Feed'>['navigation'] }) {
+  const renderPost = ({ item }: { item: Post }): JSX.Element => {
```
batter to Create a component instead of function.

usmankhalid81 commented on react-native/Kryos/src/screens/Home/HomeScreen.tsx:
```diff
@@ -0,0 +1,35 @@
+import Text from '@/components/atoms/Text';
+
+function HomeScreen({ navigation }: MainScreenProps<'Home'>) {
+  const handleLogout = () => {
+    logout().catch(console.error);
+  };
+
+  const handleNewPost = () => {
+    navigation.navigate('NewPost');
+  };
```
Try to use useCallback wherever it can be used.

usmankhalid81 commented on react-native/Kryos/src/screens/PostDetail/PostDetailScreen.jsx:
```diff
@@ -0,0 +1,197 @@
+  const [isFocused, setIsFocused] = useState(false);
+
+  // Fetch comments/replies for the post
+  const {
+    data: comments = [],
+    error: commentsError,
+    isError: isCommentsError,
+    isLoading: isCommentsLoading,
+    refetch,
+  } = useQuery({
+    enabled: !!post?.id,
+    queryFn: () => PostServices.fetchById(post.id),
+    queryKey: ['comments', post.id],
+  });
```
If we have to reuse the fetch post query, we’ll have to define it again, and if there’s a change, we’ll need to update it in all occurrences. We should move this into post-query.

usmankhalid81 commented on react-native/Kryos/src/screens/PostDetail/PostDetailScreen.jsx:
```diff
@@ -0,0 +1,197 @@
+  });
+
+  // Mutation for adding a reply
+  const addReplyMutation = useMutation({
+    mutationFn: (replyData) => PostServices.addComment(replyData),
+    onError: (error) => {
+      console.error('Error adding reply:', error);
+    },
+    onSuccess: () => {
+      // Invalidate and refetch comments and the main posts query
+      queryClient.invalidateQueries({ queryKey: ['comments', post.id] });
+      queryClient.invalidateQueries({ queryKey: ['posts'] }); // Invalidate all posts to update the count on the FeedsScreen
+      setReplyContent(''); // Clear the reply input
+    },
+  });
+
+  // Handle sending the reply
+  const handleSendReply = () => {
+    if (!replyContent.trim() || !profile?.id) return;
+
+    addReplyMutation.mutate({
+      content: replyContent,
+      parent_id: post.id,
+      user_id: profile.id,
+    });
```
Same as above.

usmankhalid81 commented on react-native/Kryos/src/screens/PostDetail/PostDetailScreen.jsx:
```diff
@@ -0,0 +1,197 @@
+    });
+  };
+
+  const renderReplies = () => {
+    if (isCommentsLoading) {
+      return (
+        <Box flex={1} justifyContent="center" alignItems="center" padding="xl">
+          <ActivityIndicator />
+          <Text marginTop="s">Loading replies...</Text>
+        </Box>
+      );
+    }
+
+    if (isCommentsError) {
+      return (
+        <Box flex={1} justifyContent="center" alignItems="center" padding="xl">
+          <Text marginBottom="m" variant="body">Error loading replies: {commentsError?.message}</Text>
+          <Button title="Retry" onPress={() => refetch()} />
+        </Box>
+      );
+    }
+
+    return (
+      <FlatList
+        data={comments}
+        keyExtractor={(item) => item.id.toString()}
```
this could be moved to separate component  i.e Replies,

usmankhalid81 commented on react-native/Kryos/src/services/instance.ts:
```diff
@@ -1,10 +1,57 @@
+instance.interceptors.request.use(
+  async (config) => {
+    // Get the session from the Supabase client to ensure we have the latest token
+    const {
+      data: { session },
+    } = await supabase.auth.getSession();
+    const accessToken = session?.access_token;
```
Get session from session service.

usmankhalid81 commented on react-native/Kryos/src/services/posts.ts:
```diff
@@ -0,0 +1,186 @@
+    }
+
+    // Validate the response directly from Supabase to match the schema
+    return validatePost({
+      content: typeof postDataResult.content === 'string' ? postDataResult.content : '',
+      created_at: typeof postDataResult.created_at === 'string' ? postDataResult.created_at : new Date().toISOString(),
+      id: typeof postDataResult.id === 'string' ? postDataResult.id : '',
+      parent: null, // Will be populated if needed
+      parent_id: typeof postDataResult.parent_id === 'string' ? postDataResult.parent_id : null,
+      replies: [], // Will be populated if needed
+      user: postDataResult.profiles as Post['user'] ?? undefined,
+      user_id: typeof postDataResult.user_id === 'string' ? postDataResult.user_id : '',
```
There are a lot of checks, in the case of a large API response, it would be hard to read and maintain. Is there any way to automate this?

usmankhalid81 commented on react-native/Kryos/src/store/slices/profileSlice.ts:
```diff
@@ -0,0 +1,64 @@
+      state.loading = false;
+      state.error = null;
+    },
+    setProfileError: (state, action: PayloadAction<null | string>) => {
+      state.error = action.payload;
+      state.loading = false;
+    },
+    setProfileLoading: (state, action: PayloadAction<boolean>) => {
+      state.loading = action.payload;
```
react query will return loading and error.

usmankhalid81 commented on react-native/Kryos/src/store/slices/uiSlice.ts:
```diff
@@ -0,0 +1,23 @@
+  isSystemLoading: true,
+};
+
+export const uiSlice = createSlice({
+  initialState,
+  name: 'ui',
+  reducers: {
+    setSystemLoading: (state, action: PayloadAction<boolean>) => {
+      state.isSystemLoading = action.payload;
+    },
+  },
```
what is the purpose of this?

usmankhalid81 reviewed (approved):
👏 👏

miankhalid committed:
feat(content): Add image upload functionality to posts

- Add react-native-image-picker for selecting images
- Implement image upload to Supabase storage
- Update Post schema and types to include image URLs
- Display images in PostItem component
- Add privacy descriptions for iOS camera/photo permissions
- Update NewPost screen with image selection UI
- Create useCreatePost hook to handle image uploads

miankhalid committed:
test: Add tests for useCreatePost hook

- Add comprehensive test coverage for useCreatePost hook
- Test initialization, state updates & post creation scenarios
- Include tests for content-only posts, image uploads & error handling
- Remove unused mock import in jest setup

miankhalid commented:
Newly implemented stuff with video:
1. Pick an image or multiple images to post
2. Upload image/s with a post
3. Display posts having image/s

https://github.com/user-attachments/assets/6e41642b-22df-4458-bf5a-2f18bbc7f98f
